### PR TITLE
Symfony 6 Support

### PR DESCRIPTION
**Pull request by tuxes3**

### CODE REVIEW

File: Controller/RestController.php [hunk 1]
```diff
@@ -14,16 +19,26 @@ class RestController extends AbstractController
 {
     use CsrfCheckerTrait;
 
+    public function __construct(
+        protected DataGridRequestHandler $dataGridRequestHandler,
```
Comment (94noni) [inline Controller/RestController.php:23]: this will not work with php 7.4, as far as composer json require < v8

Comment (bartmcleod) [reply Controller/RestController.php:23]: The policy is to have a branch per symfony version

Comment (bartmcleod) [reply Controller/RestController.php:23]: @94noni @tuxes3  Master can be used for latest and greatest, but only once the branches for older versions have been made

Comment (94noni) [reply Controller/RestController.php:23]: then this PR should update the composer require entry to have only PHP v8+ isn't it?

Comment (bartmcleod) [reply Controller/RestController.php:23]: Correct

Comment (Seb33300) [reply Controller/RestController.php:23]: IMH we should not use this syntax for now to keep compatibility with php 7.4 and keep compatibility with Symfony 5 for now

Comment (bartmcleod) [reply Controller/RestController.php:23]: @Seb33300 that would be confusing as we now have the 5.2 branch for that

Comment (Seb33300) [reply Controller/RestController.php:23]: I am not sure to understand where is the issue

**Comment by raziel057** [comment]
In this PR you need to change the composer.json to target only "php": "^8.0" as you use the syntax not compatible with php 7.4.

Maybe it could be more appropriate to first fix the deprecation for Symfony 5.4 while keeping compatibility with php 7.4. I will try to provide a PR for this this week.

**Comment by raziel057** [comment]
To keep compatibility with PHP 7.4, please replace the constructor properties with explicit properties:

In RestController:
```php
     private $dataGridRequestHandler;

    private $dataGridFormatter;

    private $translationStorage;

    private $transUnitManager;

    private $csrfTokenManager;

    public function __construct(
        DataGridRequestHandler $dataGridRequestHandler,
        DataGridFormatter $dataGridFormatter,
        StorageInterface $translationStorage,
        TransUnitManagerInterface $transUnitManager,
        ?CsrfTokenManager $csrfTokenManager
    ) {
        $this->dataGridRequestHandler = $dataGridRequestHandler;
        $this->dataGridFormatter = $dataGridFormatter;
        $this->translationStorage = $translationStorage;
        $this->transUnitManager = $transUnitManager;
        $this->csrfTokenManager = $csrfTokenManager;
    }
```

In RestController:
```php
    private $translationStorage;

    private $statsAggregator;

    private $tokenFinder;

    private $transUnitFormHandler;

    private $lexikTranslator;

    private $translator;

    private $localeManager;

    private $csrfTokenManager;

    public function __construct(
        StorageInterface $translationStorage,
        StatsAggregator $statsAggregator,
        TokenFinder $tokenFinder,
        TransUnitFormHandler $transUnitFormHandler,
        Translator $lexikTranslator,
        TranslatorInterface $translator,
        LocaleManagerInterface $localeManager,
        ?CsrfTokenManager $csrfTokenManager
    ) {
        $this->translationStorage = $translationStorage;
        $this->statsAggregator = $statsAggregator;
        $this->tokenFinder = $tokenFinder;
        $this->transUnitFormHandler = $transUnitFormHandler;
        $this->lexikTranslator = $lexikTranslator;
        $this->translator = $translator;
        $this->localeManager = $localeManager;
        $this->csrfTokenManager = $csrfTokenManager;
    }
```

File: .gitignore [hunk 2]
```diff
@@ -4,3 +4,4 @@ Propel/map
 Propel/om
 
 Tests/app/tmp/
+.idea
```
Comment (Seb33300) [inline .gitignore:7]: You should not add this to .gitignore

See https://insight.symfony.com/what-we-analyse/git.user_specific_ignored_file

File: Util/Profiler/TokenFinder.php [hunk 3]
```diff
@@ -23,7 +23,7 @@ class TokenFinder
      * @param Profiler $profiler
      * @param int      $defaultLimit
      */
-    public function __construct(Profiler $profiler, $defaultLimit)
+    public function __construct(?Profiler $profiler, $defaultLimit)
```
Comment (Seb33300) [inline Util/Profiler/TokenFinder.php:26]: Why?

**Comment by tuxes3** [comment]
> To keep compatibility with PHP 7.4, please replace the constructor properties with explicit properties:

@raziel057 done. if you see anything else let me know.

File: Util/Profiler/TokenFinder.php (hunk 3 above)
Comment (tuxes3) [reply Util/Profiler/TokenFinder.php:26]: in the `find` method it has this check:

```php
if ($this->profiler === null) {
            return [];
        }
```

therefore my logic was that thing could be null too

Comment (Seb33300) [reply Util/Profiler/TokenFinder.php:26]: Yes but this check is added by you.
Before this parameter was mandatory.

Why does it need to be optional now?

**Comment by bartmcleod** [comment]
Closing this because #413 was merged. If something was missed, let me know. I still have issues with SF 6.0.9 though, trying to fix those.

**Pull request closed without merging**